xml/ActivityDateType.xml

```diff
@@ -1,59 +1,106 @@
-<codelist name="ActivityDateType" xml:lang="en" complete="1" embedded="1">
-    <metadata>
-        <name>
-            <narrative>Activity Date Type</narrative>
-        </name>
-        <description>
-            <narrative>
-            Types of date for activities.
+<?xml version="1.0" encoding="utf-8"?>
+<codelist complete="1" embedded="1" name="ActivityDateType" xml:lang="en">
+ <metadata>
```

Comment:
Adding notes here:

* Indentation is incorrect
* Element text adds a newline, previous version did not

Are all XML files across IATI formatted in the (old diff) way? I've pushed a commit to `french-codelists` to make the indentation 4, it's close and uses regex - all whitespace is multiplied by 4 (or whatever the indentation arg is). This is an issue for some files with multiline elements.

Reply:
the new line causes issues with the way the xml text is read, removing prettify for the moment whilst we look into it

Reply:
Apologies for getting involved here – just wanted to flag one (okay two) things:

It’s maybe worth noting that the current codelist indentation isn’t consistent anyway, see e.g. [AidTypeVocabulary](https://github.com/IATI/IATI-Codelists-NonEmbedded/blob/master/xml/AidTypeVocabulary.xml) or [BudgetNotProvided](https://github.com/IATI/IATI-Codelists-NonEmbedded/blob/master/xml/BudgetNotProvided.xml) which use 2 spaces. So normalising to 2 spaces seems okay to me.

While I’m here, can I also add: As @sargentfrancesca has flags, this PR adds leading and trailing space to all text nodes. Unlike indentation width, this is a **breaking change**. For example, if you try building a reference site against this branch (e.g. by typing `@referencebot build`) you’ll see that this breaks the reference site. So it may very well break stuff for systems using these codelists.

Reply:
@andylolz No problem - appreciated! 

We are looking into alternative ways to 'prettify' the output. In this case, `BeautifulSoup` is taking more resources to 'hack' into a format we're happy with than considering alternatives.

Reply:
> considering alternatives

Might be tricky to decipher as there is a lot of cruft, but there’s some useful code snippets [here](https://github.com/IATI/IATI-Codelists-NonEmbedded/pull/51/files) w.r.t. how to do this using lxml. It even includes the equivalent hack for 4-space indentation.

The lxml API for manipulating XML is a bit ugly (I guess that’s why you’ve chosen bs4 here) but the output will be nice.